core/lib/scmi-server/scmi_clock_consumer.c

```diff
@@ -0,0 +1,274 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024, STMicroelectronics
+ */
+
+#include <assert.h>
+#include <drivers/clk.h>
+#include <drivers/clk_dt.h>
+#include <dt-bindings/scmi/scmi-clock.h>
+#include <kernel/boot.h>
+#include <kernel/panic.h>
+#include <libfdt.h>
+#include <malloc.h>
+#include <scmi_agent_configuration.h>
+#include <scmi_clock_consumer.h>
+#include <tee_api_defines_extensions.h>
+#include <trace.h>
+
+/*
+ * struct scmi_server_clock: data for a SCMI clock in DT
+ *
+ * @domain_id: SCMI domain identifier
+ * @domain_name: SCMI domain name
+ * @clock: clock to control through SCMI protocol
+ * @flags: capabilities of the SCMI clock
+ */
+struct scmi_server_clock {
+	uint32_t domain_id;
+	const char *domain_name;
+	struct clk *clock;
+	uint32_t flags;
+};
+
+/*
+ * struct scmi_clock_consumer: context of scmi_clock_consumer.c for one channel
+ *
+ * @plat_clocks: platform clocks exposed with SCMI
+ * @plat_clocks_flags: capabilities of an plat_clocks
+ * @plat_clocks_count: number of plat_clocks and plat_clocks_flags
+ * @link: link for list of scmi_clock_consumer
+ */
+struct scmi_clock_consumer {
+	struct clk *plat_clocks;
+	uint32_t *plat_clocks_flags;
+	unsigned int plat_clocks_count;
```

Comment:
To reduce indentation issues in the functions, could rename these fields to simple `clocks`, `flags` and `count`.